### ISSUE TITLE
Test Plan - Inputs

### ISSUE THREAD

**Issue by icbicket** [741ProjectFFT/FFT#16]
This may not be a problem, depending on how you implement it, but your inputs to many of your tests are from an online random number generator. Are you generating something new each time you do a test, or are you choosing one randomly generated set of data and re-using it every time you test?

Someone pointed out to me when I was designing mine that using random numbers is unpredictable and can cause you more problems later, since you are now relying on chance to point out problems (or miss them). If your test fails, you may not know why it failed, because it was some value in your set of random numbers that caused that failure. If you are regenerating a set of random numbers every time, you may simply be lucky or unlucky on each test. Maybe it would be better to have a specifically designed input data set to test, rather than using random numbers?

You could make an input file consisting of a set of added sine functions, for instance, then you will know what frequencies you should get out of the FFT and you can compare analytically.

**Comment by halonazhao** [comment]
You are right. I should use more various test values to test my functions. But I didn't use them on purpose. It's more like I wanted to say the test cases should be in similar format. They should not be the same and I don't want to keeping using the same test cases.
Random numbers are easy to generate so I chose that as my source. But definitely, if using sin or cos to generate input numbers makes more sense. But I didn't get chance to actually change it so I keep the on line random methods to match my forwording implementation.
